packages/suite/src/components/suite/modals/confirm/ConfirmDeviceScreen.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useRef } from 'react';
+import styled from 'styled-components';
+
+import { Translation, Modal } from '@suite-components';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { useDispatch } from '@suite-hooks';
+import { QrCode, QRCODE_PADDING, QRCODE_SIZE } from '@suite-components/QrCode';
+import { TrezorDevice } from '@suite-types/index';
+import { Button, ConfirmOnDevice, ModalProps, variables } from '@trezor/components';
+import { getDeviceModel } from '@trezor/device-utils';
+import { copyToClipboard } from '@trezor/dom-utils';
+import DeviceDisconnected from './Address/components/DeviceDisconnected';
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-self: center;
+`;
+
+const Value = styled.div`
+    font-size: ${variables.FONT_SIZE.NORMAL};
+    color: ${({ theme }) => theme.TYPE_DARK_GREY};
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    font-variant-numeric: tabular-nums slashed-zero;
+    margin-bottom: 20px;
+    width: 100%;
+    background: ${({ theme }) => theme.BG_LIGHT_GREY};
+    border: 1px solid ${({ theme }) => theme.STROKE_GREY};
+    border-radius: 8px;
+    word-break: break-all;
+    padding: 10px;
+    max-width: calc(${QRCODE_SIZE}px + ${QRCODE_PADDING * 2}px);
+`;
+
+const CopyButtonWrapper = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+
+const StyledModal = styled(Modal)`
+    width: unset;
+`;
+
+const StyledDeviceDisconnected = styled(DeviceDisconnected)`
+    max-width: calc(${QRCODE_SIZE}px + ${QRCODE_PADDING * 2}px);
+`;
+
+export interface ConfirmDeviceScreenProps
+    extends Pick<ModalProps, 'heading' | 'isCancelable' | 'onCancel'> {
+    device: TrezorDevice;
+    copyButtonText: React.ReactNode;
+    copyButtonDataTest?: string;
+    isConfirmed?: boolean;
+    value: string;
+    valueDataTest?: string;
+}
+
+export const ConfirmDeviceScreen = ({
```

Comment:
Sound a bit like `DevicePromptModal` to me, quite generic. What about `ConfirmValueScreen` or `ConfirmValueOnDevice`?

Reply:
Renamed to `ConfirmValueOnDevice`.